### ISSUE TITLE
Simplify roundabout navigation instructions

### ISSUE THREAD

**Issue by dvdmrtnz** [organicmaps/organicmaps#6146]
Currently the instructions when navigating a roundabout are separated into three instructions:

* "Enter the roundabout", 
* "Take the Nth exit" and 
* "Exit the roundabout", 
* with "In X meters" added in between instructions. 

This makes them too verbose. This problem will become worse when https://github.com/organicmaps/organicmaps/pull/3130 is merged and there will be names of streets mixed in with the instructions.

It would be simpler if the instructions are consolidated into a single "**At the roundabout take the _Nth_ exit**".

Here is an example:

| Screen | Current | Proposal |
|-|-|-|
| <img src="https://github.com/organicmaps/organicmaps/assets/47610359/af3ff9d0-f725-4ab2-a8d6-a481d0248ee3" width="250">  | In 200 meters enter the roundabout, then in 100 meters take the fourth exit. | In 200 meters at the roundabout take the fourth exit. |
| <img src="https://github.com/organicmaps/organicmaps/assets/47610359/c9f138c5-ae03-4242-9e7b-13b94b1c471d" width="250"> | Enter the roundabout, then in 100 meters take the fourth exit. | At the roundabout take the fourth exit. |
| <img src="https://github.com/organicmaps/organicmaps/assets/47610359/bd841a18-94d7-4ffd-bfe1-ec88803d0961" width="250"> | Exit the roundabout | - |

**Comment by zyphlar** [comment]
I like simplifying roundabouts, though I wonder if "exit" (now) is still useful. Maybe people who live near roundabouts could do a little testing and see if the final "exit" command is accurate enough to be useful?

Roundabouts near me are so small, GPS inaccuracies and timing/speeds probably can't accurately say "exit" at just the right time. But if it was possible and accurate, I often find it helpful to just be told "do it, do it now" rather than suffer that last bit of indecision (wait, is this the 3rd? Are we counting the one I came in on?)

**Comment by nach0** [comment]
Chipping in to share my experience with the beta version linked at the top of #3130 

When entering a roundabout, the TTS said something like 'enter the roundabout onto Main Street'. I would expect the enter the roundabout instruction to be just 'enter the roundabout', and having the street announced only when exiting the roundabout.

In my short test with the _TTS-street names beta_ the street name was mentioned both when entering and exiting the roundabout, I believe it is only needed when exiting.

**Comment by biodranik** [comment]
It's not reliable to announce "Exits" from roundabouts. The driver should know it in advance to plan which lane to take.

**Comment by RedAuburn** [comment]
Agreed, the exit announcement is often way too early, and the driver should be counting instead of relying on the announcement anyway.

**Comment by TobiPeterG** [comment]
> I like simplifying roundabouts, though I wonder if "exit" (now) is still useful. Maybe people who live near roundabouts could do a little testing and see if the final "exit" command is accurate enough to be useful?
> 
> Roundabouts near me are so small, GPS inaccuracies and timing/speeds probably can't accurately say "exit" at just the right time. But if it was possible and accurate, I often find it helpful to just be told "do it, do it now" rather than suffer that last bit of indecision (wait, is this the 3rd? Are we counting the one I came in on?)

In my experience, it is not accurate enough. For me, it always tells me to exit, when I already left the roundabout, so it isn't helpful. Also, it is confusing why it's telling me "then exit in 50m", I don't care how many meters I drive in the roundabout, I just want to know where to exit.

For me, a single announcement "enter roundabout in 300m and take 4th exit" would be perfect :)

**Comment by dvdmrtnz** [comment]
fork has made a small change here but really nice progress by simplifying some of the roundabout instructions

* 'In 200m enter roundabout onto Some Road, then take the 3rd exit onto Another Road' -> In 200m enter roundabout, then take the 3rd exit onto Another Road'
* 'Enter the roundabout, then in 200m take the 3rd exit' -> 'Enter the roundabout, then take the 3rd exit'

**Comment by Jean-BaptisteC** [comment]
@dvdmrtnz FIY, a comment in this issue has been edited by a main contributor of the project without your consent

**Comment by LukasMnt** [comment]
To me, "Enter the roundabout, then take the Nth exit" is too long to say and seems ridiculous. Of course, you will enter the roundabout; why not just say "At the roundabout, take the Nth exit"? And, yes, it is only two syllables less but, as a French person, it is a lot, especially when you have multiple roundabouts following each other and the TTS can't keep up.
The first proposal (https://github.com/organicmaps/organicmaps/issues/6146#issue-1916190248) was genuinely the best.

**Comment by biodranik** [comment]
@LukasMnt here are the strings, feel free to improve: https://github.com/organicmaps/organicmaps/blob/master/data/strings/sound.txt#L160

**Comment by LukasMnt** [comment]
> [@LukasMnt](https://github.com/LukasMnt) here are the strings, feel free to improve: https://github.com/organicmaps/organicmaps/blob/master/data/strings/sound.txt#L160

The problem now is that, the way it is currently implemented, I can't just change the strings and have the result I expect. I would need to deeply change the way these strings are called (a thing I wouldn't do now because of a lack of time, experience, and knowledge in the different spoken languages).
In fact, it seems that the instructions "Enter the roundabout" and "take the Nth exit" are two separate instructions plus the link word "then", while they should be a single one ("At the roundabout, take the Nth exit").
Merging these into one would make the TTS much smoother, improving the user experience in situations like the one shown in the following image.
<img width="316" height="716" alt="Image" src="https://github.com/user-attachments/assets/520a28d9-1b8b-4dcc-92cb-51fca6ca6aa4" />

**Comment by biodranik** [comment]
Won't changing "Enter the roundabout." to "At the roundabout" (no dot at the end) help without a refactoring?

**Comment by biodranik** [comment]
1. Knowing the approximate distance from the roundabout's entrance to its exit may be useful for drivers, especially for larger/longer roundabouts. Is it really unimportant information in _all_ cases to drop? Maybe simplification should be applied only for smaller roundabouts?
2. Are we talking about the case with street name announcements turned on or off? These two cases are different and may require more complex variations.

**Comment by LukasMnt** [comment]
> Won't changing "Enter the roundabout." to "At the roundabout" (no dot at the end) help without a refactoring? 

The problem with this wording is that your sentence will be "At the roundabout, then take the Nth exit" which is not really well formulated


> 1. Knowing the approximate distance from the roundabout's entrance to its exit may be useful for drivers, especially for larger/longer roundabouts. Is it really unimportant information in _all_ cases to drop? Maybe simplification should be applied only for smaller roundabouts?
> 2. Are we talking about the case with street name announcements turned on or off? These two cases are different and may require more complex variations.

I agree that the current way it is done is great for huge roundabouts. I am not looking for any simplification in the current instructions, simply trying to make them much smoother.
